brave/brave-core
# Backend Work for Crypto Wallets reset feature

Partially fixes: https://github.com/brave/brave-browser/issues/5741
Fixes: https://github.com/brave/brave-browser/issues/7716

After the deletion of the DB entry, two things have to happen:

1. All tabs containing `brave://wallet` need to close
2. Brave must restart

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
Note: The Extension at this time lacks the `"management"` permissions allowing programmatic reload, so the above snippet will not work. However, these steps can be used to verify functionality:

1. Clean profile, navigate to `brave://wallet`
2. Import/create a new wallet
3. Confirm wallet has been created.
4. In that page, open up the console.
5. Run:
```
chrome.braveWallet.resetWallet()
```
6. Crypto wallets tabs will close and Brave will restart
7. Navigate to `brave://wallet`, confirm state is fresh
8. Do the above steps then both create a new wallet and import an account and confirm this works as expected. 

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
